geth/peers/topicpool.go

```diff
@@ -64,36 +69,46 @@ func (t *TopicPool) addToPeerPool(peer *peerInfo) {
 	if _, ok := t.peerPool[peer.node.ID]; ok {
 		return
 	}
-	item := &peerInfoItem{peerInfo: peer}
-	t.peerPool[peer.node.ID] = item
-	heap.Push(&t.peerPoolQueue, item)
+	t.peerPool[peer.node.ID] = &peerInfoItem{
+		peerInfo: peer,
+		index:    notQueuedIndex,
+	}
 }
 
-func (t *TopicPool) removeFromPeerPool(nodeID discv5.NodeID) {
-	peer, ok := t.peerPool[nodeID]
-	if !ok {
-		return
-	}
-	delete(t.peerPool, nodeID)
-	heap.Remove(&t.peerPoolQueue, peer.index)
+// addToQueue cannot be called if peer is no the pool
```

Comment:
Would like to see the intention of the function as comment here and limitations as second sentence or internal comment. 

What does it mean here "cannot be called"? Would expect an error then. Or do you mean that it will never happen?

Reply:
makes sense, better to add protection instead of such vague comment